aptos-move/block-executor/src/executor.rs

```diff
@@ -1087,7 +1207,29 @@ where
                     counters::update_sequential_txn_gas_counters(&fee_statement);
 
                     // Apply the writes.
-                    Self::apply_output_sequential(&unsync_map, &output);
+                    // TODO: return code invariant error if dynamic change set optimizations disabled.
+                    Self::apply_output_sequential(&unsync_map, &output)
+                        .map_err(|e| Error::FallbackToSequential(e))?;
+
+                    let mut finalized_groups = vec![];
+                    for (group_key, group_metadata_op) in
+                        output.resource_group_metadata_ops().into_iter()
+                    {
+                        let finalized_group = unsync_map.group_to_commit(&group_key);
```

Comment:
This is identical (module TODO with the error and error message) to code on line 505 where we commit a group? Might not worth factoring out though

Reply:
yep, and moreover, likely parallel will fallback to sequential and sequential will have to treat these invariants differently.